ratbagd/ratbagd.c

```diff
@@ -204,6 +204,67 @@ static const sd_bus_vtable ratbagd_vtable[] = {
 	SD_BUS_VTABLE_END,
 };
 
+/* 32 pre-picked names. If you have more than 32 devices connected and need
+ * ratbag to control them all, congratulations, you are official a niche
+ * use-case.
+ */
+static const char *device_ids[] = {
+"mara", "capybara", "porcupine", "paca",
+"vole", "woodrat", "gerbil", "shrew",
+"hare", "beaver", "gopher", "chinchilla",
+"mouse", "rat", "rabbit", "guineapig",
+"hutia", "degu", "gundi", "acouchy",
+"nutria", "tuco-tuco", "paca", "hamster",
+"zokor", "chipmunk", "squirrel", "marmot",
+"groundhog", "suslik", "agouti", "blesmol",
+};
+
+/* bitmask telling us which ones of the above are free. */
+static uint32_t device_ids_free = (1UL << ARRAY_LENGTH(device_ids)) - 1;
+
+static inline const char *
+pick_device_id(const char *sysname)
+{
+	int index;
+
+	/* search for the first number in the sysname and where we find it,
+	 * use that as lookup index. This way we get some sort of
+	 * consistency, hidraw2 is always "rat", etc.
+	 * Totally not necessary but we might as well have some fun here.
+	 * Where this fails or picks an id already chosen, we fall back to
+	 * "choose the first one free".
+	 */
+	if (sscanf(sysname, "%*[^0-9]%d", &index) == 1 &&
+	    index >= 0 && (unsigned)index < sizeof(device_ids_free) * 8 &&
+	    !!(device_ids_free & (1 << index)))
+	    goto assign;
+
+	index = ffs(device_ids_free);
+
+	/* This can only happen if you have >32 devices, so let's not worry
+	 * about this too much */
```

Comment:
On the T450s:
```
/dev/input/event0:	Lid Switch
/dev/input/event1:	Sleep Button
/dev/input/event2:	Power Button
/dev/input/event3:	AT Translated Set 2 keyboard
/dev/input/event4:	Logitech M325
/dev/input/event5:	Integrated Camera: Integrated C
/dev/input/event6:	Video Bus
/dev/input/event7:	PC Speaker
/dev/input/event8:	ThinkPad Extra Buttons
/dev/input/event9:	HDA Intel HDMI HDMI/DP,pcm=3
/dev/input/event10:	HDA Intel HDMI HDMI/DP,pcm=7
/dev/input/event11:	HDA Intel HDMI HDMI/DP,pcm=8
/dev/input/event12:	HDA Intel HDMI HDMI/DP,pcm=9
/dev/input/event13:	HDA Intel HDMI HDMI/DP,pcm=10
/dev/input/event14:	HDA Intel PCH Dock Mic
/dev/input/event15:	HDA Intel PCH Mic
/dev/input/event16:	HDA Intel PCH Dock Headphone
/dev/input/event17:	HDA Intel PCH Headphone
/dev/input/event18:	Synaptics TM3053-004
/dev/input/event19:	TPPS/2 IBM TrackPoint
/dev/input/event20:	Logitech K750
```

Just one external keyboard+mice and I am already at 21 devices. This still gives me 10 devices left, but I wonder if we should not just cycle over the list instead of returning here.

Reply:
one solution could be what jitsi or Ubuntu is doing: prepend an adjective.

Reply:
resolving based on #634 which reduces the number of nodes we have

Reply:
*Those darn chinchillas!*